### ISSUE TITLE
works great up to the point of the generation of the new xml file

### ISSUE THREAD

**Issue by colmtroy** [denchev/Wordpress-to-placehold.it#1]
Hey there,

great idea for a script!

It works great too up to the point of the generation of the new xml file. I seem to get a binary file output rather than an xml file. My original xml file was 4MB and was processed correctly. The resulting binary file is only 4k and is not xml.

I wonder is the $content value being incorrectly reassigned to a binary image?

**Comment by denchev** [comment]
I think I had a problem with that and fixed it ... but may be the fix is only on my local machine. Will check it out.

**Comment by purethemes** [comment]
Hi, I have the same issue unfortunately. But it's great script and big timesaver

**Comment by purethemes** [comment]
Ok, found the fix, just change header from

```
header('Content-Type: application/octet-stream');
```

 to

```
header('Content-Type: text/xml');
```

and you'll probably need to remove

```
if($file['type'] != 'text/xml') {
    die('You shall not pass!');
}
```

for some reason it didn't recognize it as XML file, so I've removed that check.

**Comment by denchev** [comment]
Great! Thanks for the input. You can create a pull request if you'd like.

**Comment by ab-themes** [comment]
Hi,
I also have this problem and purethemes' solution doesn't help.. :(
It generates all the images correctly but generated xml file is also an image (i have changed extension to jpg and it is placehold.it image)
I also had this problem before and the workaround was to re-upload xml again, but now it's not working anymore...

**Comment by denchev** [comment]
Hi guys,

I'm really happy that someone showed any interest in this little piece of code. I will spend some time this weekend to fix the issue with big files and update the code.

Kind regards

**Comment by ab-themes** [comment]
This is one great piece of code, very very helpful when preparing themes for sale. Before I found it i was creating a complete new WP installation, then replacing images and exporting.. and here is just a click!
Thank you very much!!

I have found where the problem is :)

this (around line 85):

``` php
$content = file_get_contents($placeholdit);
file_put_contents($placeholdit_new_file, $content);
```

replace with this:

``` php
    $image_content = file_get_contents($placeholdit);
    file_put_contents($placeholdit_new_file, $image_content);
```

content variable was overwriting content variable that contains xml data

also i have changed a+ to w+ at this line:

``` php
    $handle = fopen($file_name . '-replaced.xml', 'w+');
```

so if file exists it gets overwritten not appended.

Once again thank you for this script, it is so helpful and time-saving :)

Cheers
